dusk/grammar.py

```diff
@@ -789,16 +789,18 @@ def reduction(
         if "init" in kwargs:
             init = self.expression(kwargs["init"])
         else:
-            # TODO: "min" and "max" are kinda stupid
+            # TODO: "min" and "max" are still kinda stupid
             # we should use something like this:
             # https://en.cppreference.com/w/cpp/types/numeric_limits/max
-            # but for double it should be 1.79769e+308
-            # FIXME: probably breaks for int
+            # the current solution is:
+            #   - appropriate for doubles
+            #   - okish for floats (may trip floating point exceptions but correct outcome)
+            #   - breaks for int (undefined behavior!)
             init_map = {
                 "sum": "0",
                 "mul": "1",
-                "min": "9" * 400,
-                "max": "-" + ("9" * 400),
+                "min": "1.79769e+308",
+                "max": "-1.79769e+308",
```

Comment:
These numbers are only approximate double precision maximums right? Are we sure there isn't a non-zero digit rounded away?

We could just use `1e400` to be sure.

I think +-inf would be fine too, since there should never be a reduction over 0 elements. As long as `max(x, <<our max>>) == x` we're good.

Hmm, I guess the issue with +-inf would be floating point exceptions. Maybe we do need a compact and exact representation. :thinking: